xbmc/platform/posix/network/NetworkPosix.cpp

```diff
@@ -148,3 +192,89 @@ CNetworkInterface* CNetworkPosix::GetFirstConnectedInterface()
 
   return pNetIf;
 }
+
+bool CNetworkPosix::PingHost(unsigned long remote_ip, unsigned int timeout_ms)
+{
+  CFileHandle fd(socket(AF_INET, SOCK_DGRAM | SOCK_NONBLOCK | SOCK_CLOEXEC, IPPROTO_ICMP));
+  if (!fd)
+  {
+    CLog::Log(LOGERROR, "socket failed: {} ({})", strerror(errno), errno);
+    return false;
+  }
+
+  int ret = setsockopt(fd, SOL_IP, IP_TTL, &TTL, sizeof(TTL));
```

Comment:
From https://man7.org/linux/man-pages/man7/ip.7.html:
>> Some BSD sockets implementations also provide an IP_RECVTTL option, but an ancillary message with type IP_RECVTTL is passed with the incoming packet.  This is different from the IP_TTL option used in Linux.
Using the SOL_IP socket options level isn't portable; BSD-based stacks use the IPPROTO_IP level.

It seems Linux is not posix compliant here, not sure at which extent you can keep a common implementation for all. Maybe you'll have to reimplement for Linux (https://stackoverflow.com/questions/16165542/is-it-possible-to-read-the-ttl-ip-header-field-when-receiving-udp-packets)

Reply:
I think I leave it as an exercise for others to implement it on their platforms :)